src/provider/states/starting.rs

```diff
@@ -20,6 +20,28 @@ impl State<PodState> for Starting {
     async fn next(self: Box<Self>, pod_state: &mut PodState, _: &Pod) -> Transition<PodState> {
         if let Some(systemd_units) = &pod_state.service_units {
             for unit in systemd_units {
+                match pod_state.systemd_manager.is_running(&unit.get_name()) {
```

Comment:
I hate that this is basically a copy from below but I don't know a better way right now either...